Dockerize the deepspeed tests.

rjsu26 opened the pull request:
- **Added Docker support**: Created Dockerfile and docker-compose.yml for containerized deployments
- **Environment improvements**: Added necessary environment variables to avoid dependency conflicts in training scripts
- **Documentation updates**: Updated Tutorial.md with virtualenv setup instructions
- **Cleanup**: Removed old output logs, result CSV files, and consolidated requirements files
- **Minor fixes**: Small adjustments to DeepSpeed configuration to run faster and complete within the example run command's timeout.

rjsu26 commented:
Eventually we will split the recipe to detect H200, B200, etc and have the right pytorch/CUDA installed.

chuanli11 commented on applications/DeepSpeed-Chat/training/run_opt-350m.sh:
```diff
@@ -1,4 +1,22 @@
 #!/bin/bash
+# --- GPU arch (B200 = SM100) ---
+export TORCH_CUDA_ARCH_LIST="10.0"
```
Can this be auto-detected like how [run_opt-13b.sh](https://github.com/LambdaLabsML/DeepSpeedExamples/pull/3/files#diff-53bfcdf7a0215423ccc7ea113642ac0bbbcc1569dcbfc9b40908d217f7dddc61) works?

chuanli11 commented on applications/DeepSpeed-Chat/training/run_opt-350m_bs24_zero0.sh:
```diff
@@ -1,5 +1,22 @@
 #!/bin/bash
+# --- GPU arch (B200 = SM100) ---
+export TORCH_CUDA_ARCH_LIST="10.0"
```
Same -- can this be auto-detected like how [run_opt-13b.sh](https://github.com/LambdaLabsML/DeepSpeedExamples/pull/3/files#diff-53bfcdf7a0215423ccc7ea113642ac0bbbcc1569dcbfc9b40908d217f7dddc61) works?

rjsu26 committed:
dockerize app; update training scripts; adjust reqs/ignores

- Add .dockerignore, .gitignore, Dockerfile, docker-compose.yml, Makefile
- Update training/.deepspeed_env and training/Tutorial.md
- Update training/run_opt-* scripts (1.3b, 13b, 350m variants)
- -Tweak training/utils/ds_utils.py and training/utils/model/model_utils.py
